packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -1497,36 +1497,38 @@ export default class Meeting extends StatelessWebexPlugin {
       // const {transceiver} = event;
       const streams = event.streams[0];
 
-      this.mediaProperties.remoteStream = streams;
-
-      Trigger.trigger(
-        this,
-        {
-          file: 'meeting/index',
-          function: 'setRemoteStream:pc.ontrack'
-        },
-        EVENT_TRIGGERS.MEDIA_READY,
-        {
-          type: EVENT_TYPES.REMOTE_AUDIO,
-          stream: MediaUtil.createMediaStream([this.mediaProperties.remoteStream.getAudioTracks()[0]])
-        }
-      );
-
-      Trigger.trigger(
-        this,
-        {
-          file: 'meeting/index',
-          function: 'setRemoteStream:pc.ontrack'
-        },
-        EVENT_TRIGGERS.MEDIA_READY,
-        {
-          type: EVENT_TYPES.REMOTE_VIDEO,
-          stream: MediaUtil.createMediaStream([this.mediaProperties.remoteStream.getVideoTracks()[0]])
-        }
-      );
-
-      if (this.mediaProperties.remoteStream.getVideoTracks()[1]) {
-        this.mediaProperties.setRemoteShare(MediaUtil.createMediaStream([this.mediaProperties.remoteStream.getVideoTracks()[1]]));
+      if (event.transceiver.mid === '0') {
+        this.mediaProperties.remoteStream = streams;
+        Trigger.trigger(
+          this,
+          {
+            file: 'meeting/index',
+            function: 'setRemoteStream:pc.ontrack'
+          },
+          EVENT_TRIGGERS.MEDIA_READY,
+          {
+            type: EVENT_TYPES.REMOTE_AUDIO,
+            stream: MediaUtil.createMediaStream([event.track])
+          }
+        );
+      }
+      else if (event.transceiver.mid === '1') {
+        this.mediaProperties.setRemoteStream(MediaUtil.createMediaStream([this.mediaProperties.remoteStream.getAudioTracks()[0], event.track]));
+        Trigger.trigger(
+          this,
+          {
+            file: 'meeting/index',
+            function: 'setRemoteStream:pc.ontrack'
+          },
+          EVENT_TRIGGERS.MEDIA_READY,
+          {
+            type: EVENT_TYPES.REMOTE_VIDEO,
+            stream: MediaUtil.createMediaStream([this.mediaProperties.remoteStream.getAudioTracks()[0], event.track])
+          }
+        );
+      }
+      else if (event.transceiver.mid === '2' && event.track) {
+        this.mediaProperties.setRemoteShare(MediaUtil.createMediaStream([event.track]));
         Trigger.trigger(
```

Comment:
If I understand correctly, now we are checking the media id of the RTCRtpTransceiver in order to fire the appropriate event. Also in the event of shareTrack we also additionally check event.track to make sure a MediaStreamTrack exists? If makes sense to me.

I just have one suggestion if you want. In case the way we trigger events changes, maybe we can not repeat that section but only include it once? In addition to using constants instead of magic strings.

Here's just a suggestion I had:

```javascript
  /**
   * Sets the remote stream on the class instance and emits and
   * event to developers
   * @param {Object} pc The remote stream peer connection
   * @returns {undefined}
   * @public
   * @memberof Meeting
   */
  setRemoteStream(pc) {
    if (!pc) {
      return;
    }
    // eslint-disable-next-line no-param-reassign
    pc.ontrack = (event) => {
      // eslint-disable-next-line no-warning-comments
      // TODO: It's possible for media to not be present
      // so we might need to either
      // A) wait until we have media flowing
      // B) trigger a second event when video is flowing
      LoggerProxy.logger.log(`Meeting:index#setRemoteStream --> ontrack event received for peerConnection: ${event}`);
      // const {transceiver} = event;
      const streams = event.streams[0];
      const MEDIA_ID = {
          AUDIO_TRACK: 0,
          VIDEO_TRACK: 1,
          SHARE_TRACK: 2
      }
      const trackMediaID = parseInt(event.transceiver.mid);
      let eventType = null;
      let eventStream = null;


      switch( trackMediaID ) {
          case MEDIA_ID.AUDIO_TRACK:
              eventType = EVENT_TYPES.REMOTE_AUDIO;
              eventStream = MediaUtil.createMediaStream([event.track]);
              this.mediaProperties.remoteStream = streams;
              break;
          case MEDIA_ID.VIDEO_TRACK:
              eventType = EVENT_TYPES.REMOTE_VIDEO;
              eventStream = MediaUtil.createMediaStream([this.mediaProperties.remoteStream.getAudioTracks()[0], event.track]);
              this.mediaProperties.setRemoteStream(MediaUtil.createMediaStream([this.mediaProperties.remoteStream.getAudioTracks()[0], event.track]));
              break;
          case MEDIA_ID.SHARE_TRACK:
              if ( event.track ) {
                eventType = EVENT_TYPES.REMOTE_SHARE;
                eventStream = this.mediaProperties.remoteShare;
                this.mediaProperties.setRemoteShare(MediaUtil.createMediaStream([event.track]));
              }
              break;
      }

      if (eventType && eventStream) {
        Trigger.trigger(
          this,
          {
            file: 'meeting/index',
            function: 'setRemoteStream:pc.ontrack'
          },
          EVENT_TRIGGERS.MEDIA_READY,
          {
            type: eventType,
            stream: eventStream
          }
        );
      }
    };
  }
```

Reply:
The only other thing that jumped out at me was that you are pulling `this.mediaProperties.remoteStream.getAudioTracks()[0]` for the `EVENT_TYPES.REMOTE_VIDEO`. Why set the stream to the audio, if the event is remote video? Thanks.

Reply:
fixing it